python/ml4ir/base/config/parse_args.py

```diff
@@ -353,6 +353,13 @@ def define_args(self):
                  "(All info after the '#' in the format [key = val]).",
         )
 
+        self.add_argument(
```

Comment:
I want to argue we could completely remove this complexity. I mean, always calculate the test (your implementation should scale and I do not expect it to make things slow). Doing so: 
- removes the need for this argument
- removes the addition of the arguments in the base relevance model and base relevance pipeline (which is good, because they are not that generic)
- Reports the p-value in the dictionaries. The user, when post-processing has all the info needed to decide on stat. significance and add the boolean column `is_stat_sig`. 

What do you think?

Reply:
@mohazahran Why don't you add this in https://github.com/salesforce/ml4ir/blob/master/python/ml4ir/applications/ranking/config/parse_args.py ?

Reply:
this comment still remains. I appreciate the hard work on this PR, but I am not convinced of the argument's need. See my first comment above. Also, what @lastmansleeping suggests, seems to be a good compromise (if we actually need to compromise).

Reply:
@balikasg  Yeah I tried Ashish's suggestion. Classification testcase fails because the evaluate function still needs `ttest_pvalue_threshold` to be passed to it. So calling it with classificationModel would crash